analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,213 @@
+
+# Display table
+arm_status_count
+```
+
+### Next we want to use the annotated CN data to define the cytoband gains/losses 
+
+We want to include cytoband and gene level calls for chromosome arms that have not been defined as a gain or loss.
+
+```{r}
+# These are the chromosome arms that have not been defined as gain or loss --
+# we want to define their cytoband/gene level status
+neutral_arm_status_df <- arm_status_count %>%
+  filter(arm_dominant == "arm_neutral")
+
+# Filter the annotated CN data to include only these chromosome arms
+consensus_seg_autosomes_df_develop <- consensus_seg_autosomes_df %>%
+  select(biospecimen_id, gene_symbol, cytoband, status) %>%
+  mutate(chromosome_arm = gsub("(p|q).*", "\\1", cytoband)) %>%
+  filter(chromosome_arm %in% neutral_arm_status_df$chromosome_arm)
+
+# Now count the cytoband level calls (for each status call) and define
+# the cytoband as that status if more than 50% of the total counts are
+# for that particular status
+cytoband_status_count <- consensus_seg_autosomes_df_develop %>%
+  count(cytoband, status) %>%
+  # Spread the data -- each row represents a unique chromosome arm
+  spread(status, n) %>%
+  replace_na(list(
+    gain = 0,
+    loss = 0,
+    amplification = 0
+  )) %>%
+  group_by(cytoband) %>%
+  # The logic here is to define the arm status based on the majority of calls
+  # in the cytoband -- if the number of calls for a specific status is
+  # responsible for more than 50% of the total calls in that cytoband, then it is
+  # used to define the cytoband status
+  mutate(
+    cytoband_dominant = case_when(
+      gain > (0.49 * (gain + loss + amplification)) ~ "cytoband_gain",
+      loss > (0.49 * (gain + loss + amplification)) ~ "cytoband_loss",
+      amplification > (0.49 * (gain + loss + amplification)) ~ "cytoband_amplification",
+      TRUE ~ "cytoband_neutral"
+    )
+  ) %>%
+  select(c("cytoband", "cytoband_dominant"))
+
+# These are the cytobands that have not been defined as gain or loss --
+# we want to define their gene level status
+neutral_cytoband_status_df <- cytoband_status_count %>%
+  filter(cytoband_dominant == "cytoband_neutral")
+
+# Display table
+cytoband_status_count
+```
+
+### Now we want to use the annotated CN data to define the gene level gains/losses 
+
+```{r}
+# Filter the annotated CN data to include only these cytobands
+consensus_seg_autosomes_df_develop <- consensus_seg_autosomes_df %>%
+  select(biospecimen_id, gene_symbol, cytoband, status) %>%
+  filter(cytoband %in% neutral_cytoband_status_df$cytoband)
+
+if (!(running_in_ci)) {
+# Now count the gene level calls (for each status call) and define
+# the gene as that status if more than 50% of the total counts are
+# for that particular status
+gene_status_count <- consensus_seg_autosomes_df_develop %>%
+  count(gene_symbol, status) %>%
+  # Spread the data -- each row represents a unique chromosome arm
+  spread(status, n) %>%
+  replace_na(list(
```

Comment:
Now since this thing is happening a third time, I think a function may be in order.

Reply:
Can call it something like `status_majority_caller` or something else.